test/run_tests_model_gen_and_load.py

```diff
@@ -0,0 +1,236 @@
+import argparse
+from genericpath import isdir, isfile
+from pathlib import Path
+import re
+import os
+import shutil
+import json
+from run_tests import convert_to_test_data, Color, NoColor
+import vowpalwabbit
+
+
+"""
+This script will use run_tests to capture all available tests to run
+Its goal is to generate models (--generate_models) for all tests that are eligible (see get_tests() for the tests that are excluded)
+by using the vw python interface and just passing in the command line of the runtest
+
+Most, if not all, runtests have the -d option specified so using the cli directly in the python interface will train a model using the file specified.
+
+After the models are generated we can run the same runtests that will load each model (--load_models) by using the same python interface,
+and that will result in the datafile being processed again
+
+The idea is that we can generate the models using one vw version and load the models in an older (or newer) vw version and check for forwards (backwards) model compatibility
+"""
+
+default_working_dir_name = ".vw_runtests_model_gen_working_dir"
+default_test_file = "core.vwtest.json"
```

Comment:
To ensure you can run this script from any directory you could use pathlib:

```suggestion
default_test_file = Path(__file__).resolve().parent / "core.vwtest.json"
```

Reply:
this path is resolved a bit later